tests/foreman/api/test_contentview.py

```diff
@@ -750,6 +751,98 @@ def test_composite_content_view_with_same_repos(self, module_org, target_sat):
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # Add some content_views to the composite view
+        self.add_content_views_to_composite(composite_cv, module_org, 2)
+        # Needs_publish should be set to True when a component CV is added/removed
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        component_cvs = composite_cv.read().component
+        # Remove a component cv, should need publish now
+        component_cvs.pop(1)
+        composite_cv.component = component_cvs
+        composite_cv = composite_cv.update(['component'])
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # add a CV that has `latest` set to true
+        new_cv = target_sat.api.ContentView().create()
+        new_cv.publish()
+        composite_cv.content_view_component[0].add(
+            data={"components": [{"content_view_id": new_cv.id, "latest": "true"}]}
+        )
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # a new version of a component cv that has "latest" - needs publish
+        new_cv.publish()
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # a component CV was changed to "always update" when ccv has old version - needs publish
+        # update composite_cv after changes
+        composite_cv = composite_cv.read()
+        # get the first CV that was added, which has 1 version
+        old_component = composite_cv.content_view_component[0].read()
+        old_component.content_view.publish()
+        old_component.latest = True
+        old_component = old_component.update(['latest'])
+        # set latest to true and see if CV needs publish
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+
+    @pytest.mark.stream
+    @pytest.mark.tier2
+    def test_check_needs_publish_flag(self, target_sat):
+        """Check that the publish_only_if_needed option in the API works as intended (defaults to
+        false, is able to be overriden to true, and if so gives the appropriate message
+        if the cvs needs_publish flag is set to false)
+
+        :id: 6e4aa845-db08-4cc3-a960-ea64fb20f50c
+
+        :expectedresults: The publish_only_if_needed flag is working as intended, and is defaulted
+            to false
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        cv = target_sat.api.ContentView().create()
+        cv.publish()
+        assert not cv.read().needs_publish
+        # Check that the default for this is false
+        assert cv.publish()
```

Comment:
Why do we need this second publish? Shouldn't we just assert the one on line 833?

Reply:
Yeah, that would work......